core/src/main/java/org/eclipse/hono/adapter/AdapterConfig.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2016 Red Hat
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Red Hat - initial creation
+ */
+
+package org.eclipse.hono.adapter;
+
+import io.vertx.core.Vertx;
+import org.eclipse.hono.config.HonoClientConfigProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Minimum configuration for protocol adapters
+ */
+@Configuration
+public abstract class AdapterConfig {
```

Comment:
My main point is: why is this class declared `abstract`? Is this to prevent Spring Boot from *always* adding it to the environment when I did not provide a subclass of it?